### PR TITLE
Added API for Products

### PR DESCRIPTION
**Pull request by maksym-lebedynets**
What was done?
All criteria with all additional scope were done.

Additional scope were done:
- YAML configuration
- getProductsList lambda
- getProductsById lambda
- Frontend integration
- async/await lambda syntax (not the callback)
- ES6 modules with Babel configuration
- Webpack
- swagger documentation
- unit tests for all Lambdas
- separate file for each Lambda
- try...catch statement for handling unexpected error in getProductsById Lambda

Link to product-service API - https://yp0ekii2cc.execute-api.eu-west-1.amazonaws.com/dev/products

LInk to FE MR - https://github.com/maksym-lebedynets/aws-nodejs-program/pull/2

### CODE REVIEW

File: product-service/getProductsList.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+import products from './products.json';
+import { headers } from './constants';
+
+export const get = async event => {
+  return {
+    headers,
+    statusCode: 200,
+    body: JSON.stringify(products),
```
Comment (ArtiFisher) [inline product-service/getProductsList.js]: Potentially JSON.stringify may throw an error, so I'd recommend to wrap it in try-catch too.

File: product-service/getProductsById.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export const getById = async event => {
+  try {
+    const { pathParameters } = event;
+    const { id: requiredProductId } = pathParameters;
+    const product = products.find(({ id }) => id === (+requiredProductId));
+
+    if (product) {
+      return {
+        headers,
+        statusCode: 200,
```
Comment (ArtiFisher) [inline product-service/getProductsById.js:15]: You may consider using some npm package with predefined status codes constants to visualize which code is used.

File: product-service/__tests__/getProductsById.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      const errorResponse = {
+        headers,
+        statusCode: 500,
+        body: expect.any(String),
+      }
+
+      expect(await getById(gatewayEvent)).toEqual(errorResponse);
+    });
+  });
+});
```
Comment (ArtiFisher) [inline product-service/__tests__/getProductsById.test.js:52]: It is a good practice to leave a new line at the end of file.

**Review by ArtiFisher — APPROVED** [review]

File: product-service/getProductsById.js (hunk 2 above)
Comment (maksym-lebedynets) [reply product-service/getProductsById.js:15]: Thanks for the suggestion.

File: product-service/__tests__/getProductsById.test.js (hunk 3 above)
Comment (maksym-lebedynets) [reply product-service/__tests__/getProductsById.test.js:52]: I don't know why but this diff doesn't show it.

File: product-service/getProductsList.js (hunk 1 above)
Comment (maksym-lebedynets) [reply product-service/getProductsList.js]: Added

**Pull request merged**